MathPortal/MathPortal/Source/Data/User.swift

```diff
@@ -89,13 +55,17 @@ class User: ParseObject {
         self.tasks = object[Object.tasks.rawValue] as? [String]
         self.dateCreated = DateTools.stringFromDate(date: dateCreated)
         self.role = (PFUser.current()?[Object.role.rawValue] as? [String])?.map({ Role.fromParseString($0) }) ?? [.undefined]
-        self.age = object[Object.age.rawValue] as? Int
         self.email = email
-        self.birthDate = object[Object.birthDate.rawValue] as? Date        
-        if let imageData = PFUser.current()?[Object.profileImage.rawValue] as? Data { profileImage = UIImage(data:imageData )}
+        self.birthDate = object[Object.birthDate.rawValue] as? Date
+        // TODO: Fix : Warning: A long-running operation is being executed on the main thread. - It happenes when saving images )the problem is image.getdata
+        if let image = PFUser.current()?[Object.profileImage.rawValue] as? PFFileObject, let imageData = try? image.getData() {
```

Comment:
You should probably just try to get an URL and then use standard caching for images.